### ISSUE TITLE
Issues running rpush init in Dockerfile

### ISSUE THREAD

**Issue by mslipenchuk267** [rpush/rpush#580]
**Describe the bug**
I am having an issue running the ```rpush init``` command in my Dockerfile for my rails 6 app.
When the Dockerfile gets to my ``` RUN bundle exec rpush init``` command I get the following error:

![Screen Shot 2020-11-18 at 1 33 51 PM](https://user-images.githubusercontent.com/25811429/99572470-b4d17a00-29a2-11eb-834d-bc12e00fecf5.png)

Where ```* Install ActiveRecord migrations? [y, n] Your response must be one of: [y, n]. Please try again.``` repeats forever.

**To Reproduce**
1. Clone my application repo: https://github.com/mslipenchuk267/parliament_notification_api
2. cd into the project directory
3. Run ```docker build -t paraliment-notification .```
4. Observe the above documented error where the Dockerfile get stuck at the rpush init command

**Expected behavior**
The Dockerfile should complete the build and not hang at the rpush init command. 

**Logs and other output**
My output when running docker build: (this shows the error)
![Screen Shot 2020-11-18 at 1 33 51 PM](https://user-images.githubusercontent.com/25811429/99572470-b4d17a00-29a2-11eb-834d-bc12e00fecf5.png)

**System configuration (please complete the following information):**
 - OS: macOS
 - OS version: 10.15.7
 - Ruby version: 2.7.1
 - Rails version: 6.0.3.4
 - Rpush version: 5.2.0

**Additional context**
I want to deploy this docker image to my kubernetes cluster.

**Comment by benlangfeld** [comment]
Why would you do `rpush init` in a Dockerfile? I think you might be misunderstanding `rpush init`, docker, or both.

**Comment by mslipenchuk267** [comment]
@benlangfeld 
Hey thank you for responding.

From my understanding rpush init just sets up the initilizer rpush file and some migrations.

I didn't originally include this command in my dockerfile and it could build, but when I tried to run the container I would get the following error:
I![Screen Shot 2020-11-18 at 1 55 08 PM](https://user-images.githubusercontent.com/25811429/99574666-ad5fa000-29a5-11eb-81c7-f8f8afd549b1.png)
I definitely have puma installed during my bundle install call, so I think it's related to rpush init not being called since I know that Dockerfile works for other rails apps i've made and dockerized.

When I start my server not in docker locally it runs fine:
![Screen Shot 2020-11-18 at 1 58 56 PM](https://user-images.githubusercontent.com/25811429/99575133-370f6d80-29a6-11eb-81bb-27e443fbb346.png)

**Comment by benlangfeld** [comment]
Run `rpush init` in the container, not in the Dockerfile.

**Comment by mslipenchuk267** [comment]
Thanks, I removed that and got a new issue which I found is actually due to some code I have in the config/intializers/rpush.rb file:
![Screen Shot 2020-11-18 at 2 48 16 PM](https://user-images.githubusercontent.com/25811429/99580099-1a2a6880-29ad-11eb-81f6-59335376871e.png)
When I comment out that code the server can start correctly (but without my rpush app made).

If include that code then I get the following error when trying to start the server:
![Screen Shot 2020-11-18 at 2 49 14 PM](https://user-images.githubusercontent.com/25811429/99580239-3cbc8180-29ad-11eb-97d4-401e472112f4.png)
The error log eventually points to my config/initializers/rpush.ru file at line 141, which is the first line of code that makes the rpush app.
![Screen Shot 2020-11-18 at 2 53 51 PM](https://user-images.githubusercontent.com/25811429/99580793-e26ff080-29ad-11eb-8d41-e552dca103ee.png)

My initializer code works for me locally.

Here is the full error log

```
Puma starting in single mode...
* Version 4.3.6 (ruby 2.7.1-p83), codename: Mysterious Traveller
* Min threads: 5, max threads: 5
* Environment: production
! Unable to load application: ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  relation "rpush_apps" does not exist
LINE 8:  WHERE a.attrelid = '"rpush_apps"'::regclass
                            ^
bundler: failed to load command: puma (/notification-backend/vendor/bundle/ruby/2.7.0/bin/puma)
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  relation "rpush_apps" does not exist
LINE 8:  WHERE a.attrelid = '"rpush_apps"'::regclass
                            ^

  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/postgresql/database_statements.rb:65:in `exec'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/postgresql/database_statements.rb:65:in `block (2 levels) in query'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/dependencies/interlock.rb:48:in `block in permit_concurrent_loads'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/dependencies/interlock.rb:47:in `permit_concurrent_loads'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/postgresql/database_statements.rb:64:in `block in query'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract_adapter.rb:722:in `block (2 levels) in log'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract_adapter.rb:721:in `block in log'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract_adapter.rb:712:in `log'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/postgresql/database_statements.rb:63:in `query'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/postgresql_adapter.rb:820:in `column_definitions'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/abstract/schema_statements.rb:114:in `columns'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/schema_cache.rb:76:in `columns'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/connection_adapters/schema_cache.rb:82:in `columns_hash'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/model_schema.rb:488:in `load_schema!'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/attributes.rb:247:in `load_schema!'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/attribute_decorators.rb:50:in `load_schema!'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/model_schema.rb:478:in `block in load_schema'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/model_schema.rb:475:in `synchronize'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/model_schema.rb:475:in `load_schema'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/model_schema.rb:352:in `columns_hash'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/dynamic_matchers.rb:60:in `block in valid?'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/dynamic_matchers.rb:60:in `all?'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/dynamic_matchers.rb:60:in `valid?'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activerecord-6.0.3.4/lib/active_record/dynamic_matchers.rb:18:in `method_missing'
  /notification-backend/config/initializers/rpush.rb:141:in `<main>'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.4.8/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:59:in `load'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/bootsnap-1.4.8/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:59:in `load'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:318:in `block in load'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:291:in `load_dependency'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/dependencies.rb:318:in `load'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/engine.rb:666:in `block in load_config_initializer'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/activesupport-6.0.3.4/lib/active_support/notifications.rb:182:in `instrument'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/engine.rb:665:in `load_config_initializer'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/engine.rb:625:in `block (2 levels) in <class:Engine>'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/engine.rb:624:in `each'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/engine.rb:624:in `block in <class:Engine>'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/initializable.rb:32:in `instance_exec'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/initializable.rb:32:in `run'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/initializable.rb:61:in `block in run_initializers'
  /usr/local/lib/ruby/2.7.0/tsort.rb:228:in `block in tsort_each'
  /usr/local/lib/ruby/2.7.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
  /usr/local/lib/ruby/2.7.0/tsort.rb:422:in `block (2 levels) in each_strongly_connected_component_from'
  /usr/local/lib/ruby/2.7.0/tsort.rb:431:in `each_strongly_connected_component_from'
  /usr/local/lib/ruby/2.7.0/tsort.rb:421:in `block in each_strongly_connected_component_from'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/initializable.rb:50:in `each'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/initializable.rb:50:in `tsort_each_child'
  /usr/local/lib/ruby/2.7.0/tsort.rb:415:in `call'
  /usr/local/lib/ruby/2.7.0/tsort.rb:415:in `each_strongly_connected_component_from'
  /usr/local/lib/ruby/2.7.0/tsort.rb:349:in `block in each_strongly_connected_component'
  /usr/local/lib/ruby/2.7.0/tsort.rb:347:in `each'
  /usr/local/lib/ruby/2.7.0/tsort.rb:347:in `call'
  /usr/local/lib/ruby/2.7.0/tsort.rb:347:in `each_strongly_connected_component'
  /usr/local/lib/ruby/2.7.0/tsort.rb:226:in `tsort_each'
  /usr/local/lib/ruby/2.7.0/tsort.rb:205:in `tsort_each'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/initializable.rb:60:in `run_initializers'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/railties-6.0.3.4/lib/rails/application.rb:363:in `initialize!'
  /notification-backend/config/environment.rb:5:in `<top (required)>'
  config.ru:3:in `require_relative'
  config.ru:3:in `block in <main>'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/builder.rb:116:in `eval'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/builder.rb:116:in `new_from_string'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/builder.rb:105:in `load_file'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/builder.rb:66:in `parse_file'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/puma-4.3.6/lib/puma/configuration.rb:321:in `load_rackup'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/puma-4.3.6/lib/puma/configuration.rb:246:in `app'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/puma-4.3.6/lib/puma/runner.rb:155:in `load_and_bind'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/puma-4.3.6/lib/puma/single.rb:98:in `run'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/puma-4.3.6/lib/puma/launcher.rb:172:in `run'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/puma-4.3.6/lib/puma/cli.rb:80:in `run'
  /notification-backend/vendor/bundle/ruby/2.7.0/gems/puma-4.3.6/bin/puma:10:in `<top (required)>'
  /notification-backend/vendor/bundle/ruby/2.7.0/bin/puma:23:in `load'
  /notification-backend/vendor/bundle/ruby/2.7.0/bin/puma:23:in `<top (required)>'
```

**Comment by mslipenchuk267** [comment]
@benlangfeld 
I found out that my intializer code wasn't working because the tables had not been migrated yet to the database.
Once I migrated the tables, I then rebuilt my docker image to have the initializer code and then it could start correctly:
![Screen Shot 2020-11-18 at 3 35 40 PM](https://user-images.githubusercontent.com/25811429/99584992-b9525e80-29b3-11eb-9e11-6f51746fac0d.png)

Now my issue is that I cannot receive iOS notifications in production, so I will check out the other issues to see If I can find a fix.

Thank you for your help.
